include/seqan3/argument_parser/validators.hpp

```diff
@@ -929,7 +929,10 @@ class regex_validator
     //!\endcond
     void operator()(range_type const & v) const
     {
-         std::for_each(v.begin(), v.end(), [&] (auto cmp) { (*this)(cmp); });
+        std::for_each(v.begin(), v.end(), [&] (auto && cmp)
+        {
+            (*this)(static_cast<option_value_type const &>(cmp));
+        });
```

Comment:
```suggestion
        for (option_value_type const & file_name : v)
            (*this)(file_name);
```
?

Reply:
This would revert the actual fix xD

Reply:
This would hide the cast, but I'm fine with this version too.

Reply:
It says so in the docs 🤷